exasol_transformers_extension/deployment/utils.py

```diff
@@ -28,3 +38,32 @@ def load_and_render_statement(template_name, **kwargs) -> str:
     template = env.get_template(template_name)
     statement = template.render(**kwargs)
     return statement
+
+
+def _download_slc_parts(tmp_dir, version):
+    for i in range(N_SLC_PARTS):
+        slc_part_name = f"{SLC_PARTS_PREFIX_NAME}" + str(i)
+        url = "/".join((GH_RELEASE_URL, version, slc_part_name))
+        response = requests.get(url, stream=True)
+        with open(Path(tmp_dir, slc_part_name), 'wb') as f:
+            f.write(response.content)
+
```

Comment:
```suggestion
def _download_slc_parts(directory, version):
    parts = (f"{SLC_PARTS_PREFIX_NAME}{index}" for index in range(N_SLC_PARTS))
    urls = ("/".join((GH_RELEASE_URL, version, part)) for part in parts)
    for url in urls:
        response = requests.get(url, stream=True)
        response.raise_for_status()
        with open(Path(directory, slc_part_name), 'wb') as f:
            f.write(response.content)
```

Reply:
loops are small but don't you think it would be more confusing to increase to 3 loops?

Reply:
A generator is lazy so it is not actually a loop in that sense. I personally find it easier and better to read and I think it separates the individual concerns better. Still for sure also partially matter of taster. Performance wise I think neither version  will be much faster. If you are concerned regarding performance I would recommend to download the parts asynchronously.

:warning: If you want to keep it as is that is fine with me, but I suggest you at least add the **_status check_** `response.raise_for_status()` .